tajo-storage/tajo-storage-hdfs/src/main/java/org/apache/tajo/storage/orc/ORCScanner.java

```diff
@@ -0,0 +1,328 @@
+      case TEXT:
+      case CHAR:
+      case PROTOBUF:
+        return new SliceVector();
+
+      default:
+        throw new UnsupportedException("This data type is not supported currently: "+type.toString());
+    }
+  }
+
+  private FileSystem fs;
+  private FSDataInputStream fis;
+
+  private static class ColumnInfo {
+    TajoDataTypes.DataType type;
+    int id;
+  }
+
+  /**
+   * Temporary array for caching column info
+   */
+  private ColumnInfo [] targetColInfo;
+
+  @Override
+  public void init() throws IOException {
+    OrcReader orcReader;
+
+    if (targets == null) {
+      targets = schema.toArray();
+    }
+
+    super.init();
+
+    Path path = fragment.getPath();
+
+    if(fs == null) {
+      fs = FileScanner.getFileSystem((TajoConf)conf, path);
+    }
+
+    if(fis == null) {
+      fis = fs.open(path);
+    }
+
+    OrcDataSource orcDataSource = new HdfsOrcDataSource(
+        this.fragment.getPath().toString(),
+        fis,
+        fs.getFileStatus(path).getLen(),
+        Integer.parseInt(meta.getOption(StorageConstants.ORC_MAX_MERGE_DISTANCE,
+          StorageConstants.DEFAULT_ORC_MAX_MERGE_DISTANCE)));
+
+    targetColInfo = new ColumnInfo[targets.length];
+    for (int i=0; i<targets.length; i++) {
+      ColumnInfo cinfo = new ColumnInfo();
+      cinfo.type = targets[i].getDataType();
+      cinfo.id = schema.getColumnId(targets[i].getQualifiedName());
+      targetColInfo[i] = cinfo;
+    }
+
+    // creating vectors for buffering
+    vectors = new Vector[targetColInfo.length];
+    for (int i=0; i<targetColInfo.length; i++) {
+      vectors[i] = createOrcVector(targetColInfo[i].type);
+    }
+
+    Set<Integer> columnSet = new HashSet<Integer>();
+    for (ColumnInfo colInfo: targetColInfo) {
+      columnSet.add(colInfo.id);
+    }
+
+    orcReader = new OrcReader(orcDataSource, new OrcMetadataReader());
+
+    // TODO: make OrcPredicate useful
+    // TODO: TimeZone should be from conf
+    // TODO: it might be splittable
+    recordReader = orcReader.createRecordReader(columnSet, OrcPredicate.TRUE,
+        fragment.getStartKey(), fragment.getLength(), DateTimeZone.getDefault());
+
+    LOG.debug("file fragment { path: " + fragment.getPath() +
+      ", start offset: " + fragment.getStartKey() +
+      ", length: " + fragment.getLength() + "}");
+
+    getNextBatch();
+  }
+
+  @Override
+  public Tuple next() throws IOException {
+    if (currentPosInBatch == batchSize) {
+      getNextBatch();
+
+      // EOF
+      if (batchSize == -1) {
+        return null;
+      }
+    }
+
+    Tuple tuple = new VTuple(targets.length);
+
+    for (int i=0; i<targetColInfo.length; i++) {
+      tuple.put(i, createValueDatum(vectors[i], targetColInfo[i].type));
+    }
+
+    currentPosInBatch++;
+
+    return tuple;
+  }
+
+  // TODO: support more types
+  private Datum createValueDatum(Vector vector, TajoDataTypes.DataType type) {
+    switch (type.getType()) {
+      case INT1:
+      case UINT1:
```

Comment:
As I mentioned above, unsigned types can be omitted because they are not supported.